clients/cli/docs/README.md

```diff
@@ -0,0 +1,149 @@
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Apache Gravitino Command Line Interface
+
+Apache Gravitino CLI is a command-line tool that interacts with the Gravitino server to manage and query entities like metalakes, catalogs, schemas, and tables. The tool provides options for listing information about Gravitino entities and in future versions support creating, deleting, and updating these entities.
+
+## Table of Contents
+
+- [Features](#features)
+- [Installation](#installation)
+- [Usage](#usage)
+- [Commands](#commands)
+- [Running Tests](#running-tests)
+- [Contributing](#contributing)
+- [License](#license)
+
+## Features
+
+- Retrieve server version
+- Provide help on usage
+- Manage Gravitino entities such as Metalakes, Catalogs, Schemas, and Tables
+- List details about Gravitino entities
+
+## Installation
+
+### Prerequisites
+
+Before you can build and run this project, it is suggested you have the following installed:
+
+- Java 8 or higher
+
+### Build the Project
+
+1. Clone the entire Gravitino repository:
+
+    ```bash
+    git clone https://github.com/apache/gravitino
+    ```
+
+2. Build the CLI sub-project using Gradle:
+
+    ```bash
+    ./gradlew :clients:cli:build
+    ```
+3. Create an alias:
+
+    ```bash
+    alias gcli='java -jar clients/cli/build/libs/gravitino-cli-*-incubating-SNAPSHOT.jar'
+    ```
+3. Test the command:
+    ```bash
+    gcli --help
+    ```
+
+## Usage
+
+To run the Gravitino CLI, use the following command structure:
+
+```bash
+usage: gcli [metalake|catalog|schema|table] [list|details|create|delete|update] [options]
+Options
+ -f,--name <arg>       full entity name (dot separated)
+ -h,--help             command help information
+ -m,--metalake <arg>   Metalake name
+ -r,--server           Gravitino server version
+ -u,--url <arg>        Gravitino URL (default: http://localhost:8090)
+ -v,--version          Gravitino client version
+```
+
+## Commands
+The following commands are available for entity management:
+
+- list: List available entities
+- details: Show detailed information about an entity
+- create: Create a new entity
+- delete: Delete an existing entity
+- update: Update an existing entity
+
+### Examples
+List All Metalakes
+
+```bash
+gcli list
+```
+
+Get Details of a Specific Metalake
+
+```bash
+gcli metalake details --metalake my-metalake
+```
+
+List Tables in a Catalog
+
+```bash
+gcli metalake list --metalake my-metalake
```

Comment:
Why metalake list still needs to specify the metalake name?

Reply:
As described in the documentation the metalake can be set in two ways, I have just chosen the first way.
1. Passed in on the command line via the `--metalake` parameter.
2. Set via the 'GRAVITINO_METALAKE' environment variable.

The user can select what to use. In the next PR they can also specify it via a configuration file.

`gcli metalake list` lists the schemas in a metalake, so the metalake needs to be specified. The command `details` shows info about an entity, the command `list` lists information about its children.

Reply:
I'm a little confused about this command, intuitively, `gcli metalake list` makes me feel like this command will list the metalakes in Gravitino, that's why I think we don't need to specify the metalake name. If we want to list the catalog, should it be like `gcli catalog list --metalake xxx`? 

Also, you mentioned that "lists the schemas in a metalake", I don't know if "schema" is a typo? From my understanding, it should be catalogs under metalake, right?

Reply:
`gcli list` will list the metalakes. Yes that is a typo, `gcli catalog list --metalake metalake_demo --name catalog_postgres` will list schemas in a catalog.

Reply:
If you would prefer that `gcli metalake list` list metalakes I can do that.

Reply:
To do this a column entity will need to be added to list the columns in a table

Reply:
It also means there is now no way to run the AllMetalakeDetails command